src/vmm/src/linux/vstate.rs

```diff
@@ -452,6 +458,8 @@ pub struct Vm {
 
     #[cfg(feature = "amd-sev")]
     pub tee_config: Tee,
+
+    pub guest_memfds: RangeMap<u64, (RawFd, u64)>,
```

Comment:
IMO we can keep track of this without needing a dependency. In my TDX branch I used a Vec of a custom type:

```rust
#[cfg(feature = "intel-tdx")]
pub struct GuestMemfdProperties {
    pub gpa: vm_memory::GuestAddress,
    pub size: u64,
    pub memfd_id: u64,
}
```

see here: https://github.com/containers/libkrun/pull/313/commits/6a50fe18ec699fc1bf66e5c44d2a73eb3824f957

Reply:
But this data structure provides ranged keys, so you don't need to add logic of iterating over guest addresses and checking if their within some region in a `Vec`. You simply supply an address, and the map itself will ensure you get the correct region. It's simpler IMO.